gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql

```diff
@@ -0,0 +1,122 @@
+/* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql */
+
+/*
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
+RETURNS VOID
+AS $$
+BEGIN
+	IF NOT EXISTS (
+		SELECT now() BETWEEN
+			CAST(substring(pg_get_expr(pr_filtered.parrangestart, pr_filtered.parchildrelid) FROM '#"%#"::%' FOR '#')

```

Comment:
original query has
```sql
select ... where ... and condition
```
you changed it to
```sql
select condition from ... where ...
```
if condition is false, then original query not exists, while new query exists!

Reply:
It appears that the 'adb_collect_table_stats_test' test couldn't catch such case.

It worked Ok because in the test there were tested only 2 partitions in 'db_files_history' - the default and the partition for the current month. Once the partition for the current month was removed, the query ran only on default partition, and it has `parrangestartincl` set to FALSE. So, the results were filtered out before checking the partition range.

Even with the query below the test was able to pass:
```
	IF NOT EXISTS ( 
		SELECT 1 
		FROM ( 
			SELECT pr.parrangestart, pr.parrangeend, pr.parchildrelid 
			FROM pg_partition pp 
			JOIN pg_class cl1 ON pp.parrelid = cl1.oid 
			JOIN pg_partition_rule pr ON pr.paroid = pp.oid 
			JOIN pg_class cl2 ON cl2.oid = pr.parchildrelid 
			JOIN pg_namespace n1 ON cl1.relnamespace = n1.oid 
			JOIN pg_namespace n2 ON cl2.relnamespace = n2.oid 
			WHERE 
				pp.paristemplate = false 
				AND 
				n1.nspname = 'arenadata_toolkit' 
				AND 
				n2.nspname = 'arenadata_toolkit' 
				AND 
				cl1.relname = 'db_files_history' 
				AND 
				pp.parkind = 'r'  --range 
				AND 
				pr.parrangestartincl = 't' 
		) pr_filtered ) 
	THEN
```

So, to avoid such situations, I've updated the 'adb_collect_table_stats_test' test as well as the query.
In the test I've added a case with an additional partition in the 'db_files_history'.